jwst/dq_init/dq_initialization.py

```diff
@@ -15,52 +15,65 @@
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
-def correct_model(input_model, mask_model):
+def correct_model(input_model, mask_model, exp_type_pref):
     """DQ Initialize a JWST Model"""
-
-    output_model = do_dqinit(input_model, mask_model)
+    output_model = do_dqinit(input_model, mask_model, exp_type_pref)
 
     return output_model
 
-def do_dqinit(input_model, mask_model):
+
+def do_dqinit(input_model, mask_model, exp_type_pref):
```

Comment:
Because we can't do a simple check on the first 3 characters of exp_type_pref, I wonder if it would be just as simple to not bother sending that value down to all lower-level routines to use as a check and instead just have all of the checks based on the type of the input_model, e.g. if isinstance(input_model, datamodels.GuiderRawModel). That way we don't have add the additional argument to all of the function calls.